### DISCUSSION TITLE
RIFE 实时补帧

### DISCUSSION THREAD

**Discussion by hooke007** [hooke007/mpv_PlayKit#123] [INFO 信息]
## 速览

> RIFE是什么？
https://github.com/megvii-research/ECCV2022-RIFE

当前使用版本为 官方 4.6 + 4.25lite + 4.26 + 4.26heavy 模型。

### 对比

为什么用rife（下图右），因为更平滑以及。。。
> 没有svpflow/mvtools（下图左）那般具有天马行空的想象力
> <img src="https://user-images.githubusercontent.com/41094733/168166932-ce3d332b-392a-4b8c-8e7b-ee210ef5d23f.jpg" width=500>

## mpv中的使用

在mpv中可通过vs滤镜实现rife补帧。

>非使用整合包的其他mpv用户，下载 vsNV 适配你的本地mpv可轻松完成部署。
>如果想从头手动部署指定的模块，参考[k7f的说明](https://github.com/hooke007/mpv_PlayKit/wiki/3_K7sfunc#1-%E4%BD%BF%E7%94%A8%E5%BC%95%E5%AF%BC)。

<s>mpv-lazy-20220515 之后的标准包版本已集成RIFE（仅NCNN版。TensorRT版需要下载增强包 `vsMega` ）</s>
mpv-lazy-20250525 之后的标准包版本已集成RIFE（仅DML版。TensorRT版需要下载增强包 `vsNV` ）

在懒人包[预设的rife脚本](https://github.com/hooke007/mpv_PlayKit/blob/main/portable_config/vs/MEMC_RIFE_DML.vpy)中有多个选项可供快速调节。我强烈建议你使用前最好去修改适配你的机器。

### 性能要求预估

官方 4.6 模型：

|  | NCNN版 | TRT版 |
| --- | --- | --- |
| 720p24fps-x2 | GTX1070ti | RTX2060 |
| 1080p24fps-x2 | RTX3070 | RTX2060 |
| 2160p24fps-x2 | RTX4090 | RTX4080 |

DirectML版性能略高于NCNN版但依旧远逊于TRT版。

性能需求概念实体化参考：
相比 ncnn ，2070super laptop 可以无预降分辨率（最多882p）顺利完成1080p24fps的倍帧，且跳转时间轴的延迟更低。
其它跑分数据 https://github.com/hooke007/mpv_PlayKit/discussions/255#discussioncomment-4685398

>[!Note]
>即使你的硬件离目标差一两档，脚本中依然有质量交换速度的选项提供。

***

### NCNN 版

（已淘汰，请使用DirectML版）通用方案，不挑显卡品牌。

https://github.com/hooke007/mpv_PlayKit/wiki/3_K7sfunc#rife_std

***

### DirectML 版

支持DX12显卡的通用方案，不挑显卡品牌。

https://github.com/hooke007/mpv_PlayKit/wiki/3_K7sfunc#rife_dml

>[!Tip]
>如果你的目标性能不足，注意进一步调低用户选项 `H_Pre` 的值。

***

### TensorRT 版

20系及以后N卡的专属方案（最低要求RTX2060）
_再往前的旧N卡没有张量核心(tersor core)，我不确定gtx1080ti的半精度推理性能是否足够_

https://github.com/hooke007/mpv_PlayKit/wiki/3_K7sfunc#rife_nv

>[!Tip]
>如果你的目标性能不足，注意进一步调低用户选项 `H_Pre` 的值。

优点是相对于 ncnn ，近似参数下具有大于150%的巨大效率提升，缺点是需要对不同的分辨率首次运行前，都要预生成一次引擎文件，此步骤视分辨率、模型、显卡性能而定，每次耗时1~10分钟不等。
<img src="https://user-images.githubusercontent.com/41094733/201232375-f75fd05a-3e65-46a8-9be4-4963276123d6.jpg" width=400>

**Comment by hzwer** [comment]
v4 模型还不能算实时模型吗 😭。

**Reply by hooke007** [reply]
我的瞎玩行为居然被开发者本人注意到了.......
怎么说呢，对大多数人的硬件来说还是太沉重了。（虽然vs不算一个高效的框架）如果模型能进一步精简，在当前的条件下，3060laptop这级别的显卡能无预降分辨率来实时播放1080p那就更好了

**Reply by oblessnoob** [reply]
> 我的瞎玩行为居然被开发者本人注意到了....... 怎么说呢，对大多数人的硬件来说还是太沉重了。（虽然vs不算一个高效的框架）如果模型能进一步精简，在当前的条件下，3060laptop这级别的显卡能无预降分辨率来实时播放1080p那就更好了

我之前用第一版支持tensor core的ncnn版在3060laptop上测试4.0处理1080p的帧序列，输入序列的速度差不多是29fps出头，如果把96eu也加进来的话能到31fps，正好卡在了这条线上，如果输入是24fps的动画我感觉应该就没啥问题了

**Reply by hooke007** [reply]
嘛，，，直接用nihui的ncnn那版肯定比现在更快。这里vs卡了一下，mpv的vs滤镜卡了第二下（即使vs中不做任何处理input完直接output也有明显的效率折损）。当然这不是rife的问题，不过当前pc用于实时播放也几乎只有这条途径了

**Reply by M1saka10010** [reply]
> 我的瞎玩行为居然被开发者本人注意到了....... 怎么说呢，对大多数人的硬件来说还是太沉重了。（虽然vs不算一个高效的框架）如果模型能进一步精简，在当前的条件下，3060laptop这级别的显卡能无预降分辨率来实时播放1080p那就更好了

3070laptop可以无预降1080（吃到80-90%

**Reply by iforgiveu** [reply]
RIFE这个补帧还是太吃性能了，用我小小的3060只能让视频变得卡顿，还是老老实实用mvt了

**Reply by hooke007** [reply]
3060跑trt不是问题

**Comment by natural-harmonia-gropius** [comment]
https://github.com/hooke007/MPV_lazy/blob/5015125748d619662f2f8ce69cfc08cda7554022/portable_config/vs/rife-2x.vpy#L43
我这边测下来 format 还原而不是 444p16 性能更好，带的动 1080p30 了
range 也可以合并到一起
```
format = clip.format.id
range = clip.get_frame(0).props._ColorRange
clip = clip.resize.Bicubic(format=vs.RGBS, matrix_in=1, range_in=range)
clip = clip.resize.Bicubic(format=format, matrix=1, range=1-range)
```
用[这个帖子](https://www.svp-team.com/forum/viewtopic.php?id=6582)里米奇的脚本测下来失真也更低
注意 Y Plane 下面第二行要传参进去
```
expr = 'x y - '.format(neutral=1 << (input_YUV.format.bits_per_sample - 1))
Diff = core.std.Expr([input_YUV, clip_YUV], [expr])
```
mpv 的 vapoursynth=file:buffered-frames:concurrent-frames
buffered-frames 设为 1 即可，对于补帧来说只需要用到两帧
concurrent-frames 默认等于逻辑处理器数量，我测下来等于 1.5 倍逻辑处理器数量时一些默认会掉帧复杂场景能处理的过来了（Haswell 4核无超线程 svp to 60，Whiskey Lake 6核开超线程 svp to 120，Vermeer 12核开超线程 rife x2 + svp 120）。不过不是很理解为什么，这里应该只是解码消耗，特别是24个逻辑处理器也有这种现象，很怪异

**Reply by hooke007** [reply]
> 我这边测下来 format 还原而不是 444p16 性能更好

确实是420p8的性能要求远低于444p16，当然前提是别用krig这种着色器把差距给拉回来了。。。这里我再改改吧

```
clip = clip.resize.Bicubic(format=vs.RGBS, matrix_in=1, range_in=range)
clip = clip.resize.Bicubic(format=format, matrix=1, range=1-range)
```

你这边算反了，这样yuv limited的源动态被压缩了，而yuv full的源动态反而被扩展了。

> vapoursynth=file:buffered-frames:concurrent-frames

我这里的性能依然对这两参数不敏感。。。我找个文档里提一嘴吧，让他们自行测试

**Reply by natural-harmonia-gropius** [reply]
我这边krig消耗低于vs里输出444p16
vp9 1080p30 的短片 vs 用 resize.Bicubic 能跑 1400fps 而 krig 能跑 2100 帧
```
range = clip.get_frame(0).props._ColorRange
clip = clip.resize.Bicubic(format=vs.RGBS, matrix_in=1, range_in=range)
clip = clip.resize.Bicubic(format=vs.YUV444P16, matrix=1, range=1-range)
clip.set_output()
```
https://github.com/hooke007/MPV_lazy/blob/5015125748d619662f2f8ce69cfc08cda7554022/portable_config/vs/rife-2x.vpy#L44-L45
1-0=1 ，limited 实测没反，没有 full 的样本

**Reply by hooke007** [reply]
我觉得我可能理解有误于是拿你仓库里的脚本测了，动态确实错的。你再检查一下....
![Snipaste_2022-05-21_13-41-05](https://user-images.githubusercontent.com/41094733/169651929-58f6b232-31a9-444f-83fb-b9981de4d3f0.jpg)

**Reply by hooke007** [reply]
这里错了，不能指定range_in
https://github.com/Natural-Harmonia-Gropius/mpv_config/blob/a350ad01c8f7a46eb571fc207b269db51909492d/filters/interp.py#L131

**Reply by hooke007** [reply]
另外一点发现就是这里不同的解码差异巨大 dxva2-copy≈d3d11va-copy＞＞nvdec-copy≈no，1080_420p8用nvdec-copy/no时我完全没有余力去跑别的着色器。

**Reply by hooke007** [reply]
你那的r7测下来有没有这个问题
https://github.com/HomeOfVapourSynthEvolution/VapourSynth-RIFE-ncnn-Vulkan/issues/35

**Reply by natural-harmonia-gropius** [reply]
会崩

**Reply by natural-harmonia-gropius** [reply]
你试过vseditor跑vfr的视频吗，mpv传到vs里总是vfr的数据

**Reply by hooke007** [reply]
vse跑vfr也不崩

**Reply by natural-harmonia-gropius** [reply]
[vsh::muldivRational(&d->factorNum, &d->factorDen, d->vi.fpsDen, d->vi.fpsNum);](https://github.com/HomeOfVapourSynthEvolution/VapourSynth-RIFE-ncnn-Vulkan/blob/60b7e1661d74cbfdf1c32d87456fc7a1e84bef4b/RIFE/plugin.cpp#L219)
我感觉是这行报的错，`clip.fps_den` 照文档说的应该是0

**Comment by 1VeniVediVeci1** [comment]
3070笔记本用了trt可以进行1080p实时倍频了，非常舒适，不过感觉原来的代码有点问题，个人是修改了2倍频版本的vs脚本才用起来的。

**Reply by hooke007** [reply]
根据前面的反馈3070laptop 即使用NCNN也不需要降分 https://github.com/hooke007/MPV_lazy/discussions/123#discussioncomment-2798125 问题是指什么？

**Reply by 1VeniVediVeci1** [reply]
ncnn那个可能是因为我没改输出格式，换成source的话也许可以1080p。cuda的脚本我从git上重新copy了一下，可以用，可能是我当时修改的时候文本格式出现了点错误。

**Reply by hooke007** [reply]
source通常只可能会增加性能压力而不是减少...

**Reply by 1VeniVediVeci1** [reply]
我回到宿舍后重新研究了一下，确实和输出格式无关。其实是当时测试的时候出了个bug，视频的尺寸是2592*1080，ncnn版本不能正常倍频，trt版本可以。如果是1920*1080的视频，3070-laptop无论是ncnn还是trt都是可以正常倍频的。

**Reply by ww2283** [reply]
想问一下 trt的话windows环境中需要系统预先装好CUDA和TensorRT吗？

**Reply by hooke007** [reply]
> TensorRT版的一键部署包将以独立扩展包的形式，不定期与懒人包一起发布于release（插件库的体积过大，不适合整合发布）

这句话应该没有歧义，如果涉及到外部安装的东西也无法进行便携化处理，更不会有一键包这种东西

**Reply by ww2283** [reply]
感谢！我看到生成引擎文件的时候提示了一句话， 大意是参考TensorRT官方配置预配置好的话可以调用更大的库。一闪就过去了，没来得及记下来。总之目前用起来TensorRT版确实非常流畅。

**Reply by hooke007** [reply]
这话也没错，上游实现的时候已经最大化精简了依赖（牺牲了一点性能）

**Reply by ww2283** [reply]
提个小建议，vs脚本能不能加个2倍或2.5倍的帧率判定。因为有时候输入视频是24fps的，2倍就有点奇怪。

**Reply by ww2283** [reply]
> 这话也没错，上游实现的时候已经最大化精简了依赖（牺牲了一点性能）

好的，那等下一个版本的时候我再试试给CUDA和TensorRT装上看有多大提升。目前我是一张2080Ti, RIFE 4.6+FSRCNNX+Krig非常流畅。GPU算力和显存都有余量。

**Reply by 1VeniVediVeci1** [reply]
> 感谢！我看到生成引擎文件的时候提示了一句话， 大意是参考TensorRT官方配置预配置好的话可以调用更大的库。一闪就过去了，没来得及记下来。总之目前用起来TensorRT版确实非常流畅。

有相关提示吗？我这边生成好像没有

**Reply by hooke007** [reply]
并不奇怪啊（大多数视频都是24(23.976)FPS），目的是流畅，实际达到40+FPS的时候肉眼感官已经很好了，没有必要追求补帧到60（显示器的实际刷新也基本不是整数60）。
display-resample会完善最终阶段的输出采样匹配刷新率。
ncnn做2.x倍和3倍几乎是一样的性能开销（我猜内部始终是做到3倍然后抽帧或者采样处理）

**Reply by ww2283** [reply]
> > 感谢！我看到生成引擎文件的时候提示了一句话， 大意是参考TensorRT官方配置预配置好的话可以调用更大的库。一闪就过去了，没来得及记下来。总之目前用起来TensorRT版确实非常流畅。
> 
> 有相关提示吗？我这边生成好像没有

就是当时跳出来的那个cmd窗口有一行warning，很快就过去了。确实也不影响使用，这个版本我已经非常满足了，等下个版本再折腾一下。

**Reply by ww2283** [reply]
> 并不奇怪啊（大多数视频都是24(23.976)FPS），目的是流畅，实际达到40+FPS的时候肉眼感官已经很好了，没有必要追求补帧到60（显示器的实际刷新也基本不是整数60）。 display-resample会完善最终阶段的输出采样匹配刷新率。 ncnn做2.x倍和3倍几乎是一样的性能开销（我猜内部始终是做到3倍然后抽帧或者采样处理）

hmmm，有道理

**Reply by 1VeniVediVeci1** [reply]
> > > 感谢！我看到生成引擎文件的时候提示了一句话， 大意是参考TensorRT官方配置预配置好的话可以调用更大的库。一闪就过去了，没来得及记下来。总之目前用起来TensorRT版确实非常流畅。
> > 
> > 
> > 有相关提示吗？我这边生成好像没有
> 
> 就是当时跳出来的那个cmd窗口有一行warning，很快就过去了。确实也不影响使用，这个版本我已经非常满足了，等下个版本再折腾一下。

那个warning应该是cuda11.7以后新增的一个添加环境变量的warning，可以减少内存占用，不影响推理速度

**Reply by ww2283** [reply]
> > > > 感谢！我看到生成引擎文件的时候提示了一句话， 大意是参考TensorRT官方配置预配置好的话可以调用更大的库。一闪就过去了，没来得及记下来。总之目前用起来TensorRT版确实非常流畅。
> > > 
> > > 
> > > 有相关提示吗？我这边生成好像没有
> > 
> > 
> > 就是当时跳出来的那个cmd窗口有一行warning，很快就过去了。确实也不影响使用，这个版本我已经非常满足了，等下个版本再折腾一下。
> 
> 那个warning应该是cuda11.7以后新增的一个添加环境变量的warning，可以减少内存占用，不影响推理速度

哦好的，多谢！那就省事了～目前内存大大的富裕，不折腾了

**Reply by lgthappy** [reply]
> 3070笔记本用了trt可以进行1080p实时倍频了，非常舒适，不过感觉原来的代码有点问题，个人是修改了2倍频版本的vs脚本才用起来的。

我也刚刚看到，你的代码方便发出来吗？转换及补帧的代码，我很想学习，但不太会

**Comment by ghost** [comment]


**Reply by hooke007** [reply]
以前就可以，只有极个别原生滤镜会自动触发autoconvert帮你做（像素格式）转换。跑软解或者copy就不需要autoconvert。

**Comment by catshitz** [comment]
2023年了，SVP4Pro的RIFE-Tensor带4K还是有点吃力（看来不上4090不行）😅

**Reply by hooke007** [reply]
一样是来自上游的方案，没有区别（2160p24fpsx2 4080就足够了

**Reply by microka** [reply]
大佬，请问如果是4K原盘24fps补帧到48fps，4080足够吗？有没哪些场景是4080仍吃力需要上4090才能满足的？（手持Intel UHD750核显还没入独显

**Reply by hooke007** [reply]
码率无关。但是4.6之后性能消耗已开始大幅上升

**Reply by microka** [reply]
所以是能上4090会更好吗？😂

**Comment by letterk** [comment]
<img width=300 src=https://github.com/hooke007/MPV_lazy/assets/9822709/41ed5a04-0762-433b-bc3d-590f8a74571b>
我这里怎么rife的瑕疵率高于mvtools
画面经常抽搐畸变
伪影rife确实比较好

**Reply by hooke007** [reply]
忘了记录了，密集的网格线条是重灾区

**Reply by letterk** [reply]
还有某些字幕直接在视频里的, 比如这个b站上的电影, 字幕都在抖
是对线条支持不好吗

**Reply by oblessnoob** [reply]
硬字幕这种在运动物体上面叠一层字的情况属于细小物体运动问题，目前绝大多数光流算法和补帧算法都不能很好地解决这个问题

**Comment by yueisme** [comment]
我在使用 2023EX 版本的`MEMC_RIFE_STD.vpy`，GPU 1080ti 流畅播放1080p及以上视频大概需要改成`H_Pre = 900`以内

HDR映射SDR时，画面整体亮度高的镜头（比如天空）有闪烁的现象

**Reply by hooke007** [reply]
无关rife
vs滤镜不参与tonemap
另 1080可以用trt

**Reply by yueisme** [reply]
2024v0 vsMega版本的`MEMC_RIFE_NV.vpy`启动报错

<details>

<summary>错误日志</summary>

```text
[02/25/2024-21:09:47] [I] [TRT] Global timing cache in use. Profiling results in this builder pass will be stored.
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x00000000000201ac due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x00000000020401ac due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x00000000020601ac due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x00000000020801ac due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x00000000020a01ac due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x00000000020c01ac due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x0000000000020279 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x0000000002040279 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x0000000002060279 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x0000000002080279 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x00000000020a0279 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x00000000020c0279 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x0000000000020628 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x0000000002040628 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x0000000002060628 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x0000000002080628 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x00000000020a0628 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[9]: Skipping tactic 0x00000000020c0628 due to exception no kernel image is available for execution on the device
[02/25/2024-21:09:53] [E] Error[10]: Could not find any implementation for node ConvTranspose_42.
[02/25/2024-21:09:53] [E] Error[10]: [optimizer.cpp::nvinfer1::builder::cgraph::LeafCNode::computeCosts::4048] Error Code 10: Internal Error (Could not find any implementation for node ConvTranspose_42.)
[02/25/2024-21:09:53] [E] Engine could not be created from network
[02/25/2024-21:09:53] [E] Building engine failed
[02/25/2024-21:09:53] [E] Failed to create engine from model or file.
[02/25/2024-21:09:53] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec [TensorRT v9200] # C:/Users/yue/Downloads/mpv-lazy-20240102-vsMega/mpv-lazy/vapoursynth64/plugins\vsmlrt-cuda\trtexec --onnx=C:/Users/yue/Downloads/mpv-lazy-20240102-vsMega/mpv-lazy/vapoursynth64/plugins\models\rife\rife_v4.6.onnx --timingCacheFile=C:/Users/yue/Downloads/mpv-lazy-20240102-vsMega/mpv-lazy/vapoursynth64/plugins\models\rife\4490c10d.engine.cache --device=0 --saveEngine=C:/Users/yue/Downloads/mpv-lazy-20240102-vsMega/mpv-lazy/vapoursynth64/plugins\models\rife\4490c10d.engine --minShapes=input:1x11x256x320 --optShapes=input:1x11x1088x1920 --maxShapes=input:1x11x2176x4096 --fp16 --tacticSources=-CUBLAS,-CUBLAS_LT,-CUDNN,+EDGE_MASK_CONVOLUTIONS,+JIT_CONVOLUTIONS --useCudaGraph --noDataTransfers --noTF32 --inputIOFormats=fp16:chw --outputIOFormats=fp16:chw --layerPrecisions=*:fp16 --layerOutputTypes=*:fp16 --precisionConstraints=obey --builderOptimizationLevel=3

```

</details>


2023EX vsMega版本能正常开启，但是性能差到无法使用🥲

**Reply by yueisme** [reply]
> vs滤镜不参与tonemap

对比过开关RIFE，确实是有低强度的闪烁，应该是模型的瑕疵？

**Reply by hooke007** [reply]
亮度数据一种由mpv实时测量，另一种由上一帧复制，无论如何与rife无关。
帧率和画面都改变了，但mpv没有对应的正确处理，不是rife的问题。

这个问题建议调节hdr相关的选项。

**Reply by hooke007** [reply]
> 启动报错

<s>可能和fp16有关，</s>trt提升了最低硬件要求，最后一个兼容的n卡10系的版本应该是2023V1

edit: 2025.04.19 目前 `RIFE_DML` 已经跑通测试，非RTX的n卡可选DirectML版本作为替代。

**Comment by sf467** [comment]
目前没有找到在 linux 下使用 tensorRT 的 RIFE 补帧方法

**Reply by tanyiok1234** [reply]
这还真不是，比如我在arch下试过是可以的，你需要AUR的vapoursynth-plugin-mlrt-trt-runtime-git包。。。其它系统嘛。。。加油

**Reply by sf467** [reply]
又折腾了折腾，当时确实安装的这个包，也就顺便开始装 tensorrt, 
因为 makepkg 的工作目录我定在了 /tmp 里导致编译 tensorrt 的时候内存满了，编译没成功。当时我以为这包是坏的（

**Reply by sf467** [reply]
> 这还真不是，比如我在arch下试过是可以的，你需要AUR的vapoursynth-plugin-mlrt-trt-runtime-git包。。。其它系统嘛。。。加油

另外我看到 AUR 上这个包上标了一个 8vram ，是不是显存比这个小的就用不了了

**Reply by tanyiok1234** [reply]
> > 这还真不是，比如我在arch下试过是可以的，你需要AUR的vapoursynth-plugin-mlrt-trt-runtime-git包。。。其它系统嘛。。。加油
> 
> 
> 
> 另外我看到 AUR 上这个包上标了一个 8vram ，是不是显存比这个小的就用不了了

我觉得不见得，看实际用什么模型。。。不过我没有小于8G的卡了，你还是直接试一试最快

**Reply by hooke007** [reply]
ws_size 可以限制显存用量

**Comment by GoldJohnKing** [comment]
有个奇怪的问题。
我用2024V3版本，在7900XTX上，只要启用了RIFE_STD，一暂停mpv就崩了。
但是换成4090，还是用2024V3的RIFE_STD，就没这个问题……😅
迷惑。

**Reply by hooke007** [reply]
这些项目几乎没人会用a卡的，会崩不奇怪，大概率没人测试a卡

**Reply by GoldJohnKing** [reply]
我发现了，MEMC_RIFE_STD.vpy脚本里，Skip = True，就会在点进度条、暂停或者播放结束的时候崩掉。False就没问题。😂
我是不是该报给[上游](https://github.com/hooke007/MPV_lazy/wiki/3_K7sfunc#rife_std)去？

**Reply by hooke007** [reply]
很难汇报，要用[vseditor](https://github.com/YomikoR/VapourSynth-Editor)做单脚本复现，否则你无法确认这个问题点在mpv的接口还是vs的接口还是插件的问题

**Comment by CANNONGH** [comment]
If I set H_Pre to 1080 and perform frame interpolation on a 4K video using RIFE, will the output quality be reduced to Full HD?

**Reply by hooke007** [reply]
Sure, it dscales to 1080p firstly and then does interpolation.

**Comment by skiqing** [comment]
<img width="300" alt="MPV-00_02_30 692-N0001" src="https://github.com/user-attachments/assets/95838259-6a1b-4997-8351-8ccbb0a82459" />
心血来潮突然测试了下4K（H_Pre =1440）视频我6800XT部分可以带动。。。

**Reply by hooke007** [reply]
2k补帧的数据没有量化测试过，这个等级的显卡能用也不奇怪

**Comment by o5-null** [comment]
有没有可能使用intel npu进行RIFE加速

**Reply by hooke007** [reply]
理论上有没测试过，接口在vsmlrt里

**Reply by o5-null** [reply]
在github上看到个可能有用的项目，问了一下作者，回我了
https://github.com/idocinthebox/Advanced-Tape-Restorer/issues/3#issuecomment-4004249164
不过这是DirectML啊。。。

**Reply by hooke007** [reply]
这项目的ai含量太高了连回复都是全ai的，纯浪费时间

**Reply by o5-null** [reply]
屎完了。
我自己写个迁移试试吧

**Reply by o5-null** [reply]
上游vs-mlrt的openvino支持又老又有依赖问题，很难编译，要不要考虑直接使用python而不使用mlrt里面的自行编译cpp库方案

**Reply by hooke007** [reply]
在我看来，npu那点算力没有考虑的意义。。。

**Reply by o5-null** [reply]
先写个benchmark测一下吧。。这玩意对ptl很有用，要是npu能支撑rife运行，就算是lite也能用

**Reply by o5-null** [reply]
测试完了，结果是能跑，但是有性能问题。arc 140v npu上运算速度在33fps左右，性能瓶颈是gridsample，尝试使用GPU XMX加速也没改善，因为频繁的精度转换反而更慢了。现在在考虑重新训练一个rife版本，使用更通用的算子来吃到更多性能加速。这一点在tensorRT上应该也是一样的，只不过力大砖飞所以速度快

**Reply by hooke007** [reply]
这和DML速度相差多少

**Reply by o5-null** [reply]
> 这和DML速度相差多少

dml应该是20多，加上考虑输出同步性能还得差一点

**Comment by skysunny2062** [comment]
有个奇怪的问题 萤幕=60hz 使用RIFE补到倍祯=48FPS  video-sync=display-resample 
那interpolation=no or yes在此环境下还有差别吗
眼盲看不出差异 也找不到方式测

**Reply by hooke007** [reply]
当然有差异，interpolation主要以混合前后两帧的方式来模糊形成“平滑”过渡。当然前提是它确实在工作

**Comment by o5-null** [comment]
对目前主流的的插帧模型进行了测试
按大致的模型效果排序
均使用2倍生成（因为最新的几个模型暂未支持可调倍率
<details>

IFUNet

https://github.com/user-attachments/assets/5dd514db-d1a8-422d-9143-9f1896999a00

IFRnet

https://github.com/user-attachments/assets/480ebf73-b959-46e3-b8c2-75c6f5e1ade2

GMSFF

https://github.com/user-attachments/assets/51a5663d-460f-4d01-b1a6-e5ac9ad32bfb

rife49

https://github.com/user-attachments/assets/83a0b054-a6be-4a8a-8f76-024345b8d221

M2M

https://github.com/user-attachments/assets/5baf13d9-9faa-4e24-b29e-d6bdee1cbec2

ATM

https://github.com/user-attachments/assets/6d5769e2-2891-438a-a605-28cc7dca2546

FLAVR

https://github.com/user-attachments/assets/f631ef21-f720-438c-8f4f-4ea09db38251

STMFNet

https://github.com/user-attachments/assets/73537742-4443-45dd-bddc-0254f48385b8

AMT

https://github.com/user-attachments/assets/050235ec-ad31-4232-9d90-d7cb2e7763be

FILM

https://github.com/user-attachments/assets/42256922-269a-4d29-9d0d-72f89b495527


目前相对来说最好的是AMT模型

<img width="1274" height="748" alt="image" src="https://github.com/user-attachments/assets/962ac1b1-22e5-4349-bceb-7b4d727d025a" />

注意到图像中大跨度变化并未出现RIFE的伪影和断裂，仅为模糊，RIFE在上下野捕获不到的时候会展现出底层光流法的扭曲痕迹

<img width="1274" height="748" alt="image" src="https://github.com/user-attachments/assets/c438f03f-2e4c-4888-a8c0-4ad45048c748" />

<img width="1274" height="748" alt="image" src="https://github.com/user-attachments/assets/deb07289-5dea-43d4-bcc0-69bb0bd9e9c6" />

</details>

**Reply by hooke007** [reply]
rife49算不上主流吧，rife里也是很偏门的了。不过我不关注力大砖飞的领域，只看实时效率

**Comment by o5-null** [comment]
使用自定义算子替换了intel上推理库里的光流推理算子
rife 4.26lite

| 指标 | Default | Custom | 改善 |
|------|---------|--------|------|
| 总耗时 | 74.32 ms | 57.48 ms | **1.29x 加速** |
| FPS | 13.46 | 17.40 | +29% |
| GridSample 耗时 | 18.56 ms | 4.20 ms | **4.42x 加速** |

现在瓶颈存在精度转换和conv上面,得修改模型架构

**Reply by f451496690shiyi** [reply]
能问下amd的话也可以这样加速rife吗？

**Reply by o5-null** [reply]
> 能问下amd的话也可以这样加速rife吗？

这个加速是通用的，在使用了intel的推理库后使用自定义算子以规避算子库中grid代码优化不足的问题后能够实现提速0.3倍，其他平台也可以使用自定义算子来吃到这个提升

**Reply by f451496690shiyi** [reply]
谢谢解答，如果方便的话可以告诉我怎么使用自定义算子吗？

**Reply by o5-null** [reply]
> 谢谢解答，如果方便的话可以告诉我怎么使用自定义算子吗？

目前没有vs可用，我只在intel openvino里实现了注入

**Reply by f451496690shiyi** [reply]
谢谢！我试着查查看怎么使用

**Comment by beefjerky331** [comment]
2026CJ版本vsNV补丁包，控制台提示 补帧RIFE_RTX(DLC-vsNV) 补帧DRBA_RTX(DLC-vsNV) 这两个VF滤镜开启失败，日志如下，意思是要更新显卡驱动么？572.83 cuda12.8居然不能支持！？
<img width="1547" height="800" alt="1" src="https://github.com/user-attachments/assets/dfa3edbb-14c5-44ad-8662-02c980d7aacf" />
<img width="640" height="209" alt="1-2" src="https://github.com/user-attachments/assets/aed9ff4c-7344-4d41-b740-2a59b400b168" />

**Reply by hooke007** [reply]
预配的包基于cuda13

**Comment by f451496690shiyi** [comment]
有可能给amd卡加速吗？（rocm？migx？）谢谢！

**Reply by f451496690shiyi** [reply]
我是9060xt，rocm7.2已经安装好了

**Reply by hooke007** [reply]
跟你本机装啥没关系，工具链目前暂不支持

**Reply by f451496690shiyi** [reply]
> 跟你本机装啥没关系，工具链目前暂不支持

我看上面有人用Intel npu加速rife我还以为amd的卡可以用同样的方法来加速rife

**Reply by o5-null** [reply]
> > 跟你本机装啥没关系，工具链目前暂不支持
> 
> 我看上面有人用Intel npu加速rife我还以为amd的卡可以用同样的方法来加速rife

可以，不过你需要自行解决vs-mrlt的问题。虽然mpv-playkit使用的上游库是有多后端支持的但是k7s并没有添加多后端支持，你得考虑换成直接使用上游库

**Reply by hooke007** [reply]
mlrt不支持，他得自己解决更底层的问题

**Reply by o5-null** [reply]
> mlrt不支持，他得自己解决更底层的问题

不是有vsmigx后端吗

**Reply by hooke007** [reply]
https://github.com/AmusementClub/vs-mlrt/issues/154
https://github.com/AmusementClub/vs-mlrt/discussions/136

**Reply by o5-null** [reply]
> [AmusementClub/vs-mlrt#154](https://github.com/AmusementClub/vs-mlrt/issues/154) [AmusementClub/vs-mlrt#136](https://github.com/AmusementClub/vs-mlrt/discussions/136)

合着全平台的gridsample都有问题啊。

**Reply by o5-null** [reply]
> > 跟你本机装啥没关系，工具链目前暂不支持
> 
> 我看上面有人用Intel npu加速rife我还以为amd的卡可以用同样的方法来加速rife

理论上可以，经过讨论后发现现在社区仍然没有可以运行的方案。也许未来会有一个修改版的rife实现更优良的性能（等吧，没空写

**Reply by f451496690shiyi** [reply]
那就只能等了，谢谢！

**Comment by o5-null** [comment]
我们有官方qq讨论群之类的吗，正在写一个允许多后端多平台的插帧模型框架，目前实现了cuda和intel xpu上的运行。这个项目是为了以后对rife进行gridsample进行优化设计的基层框架。现在需要一些建议和讨论

**Reply by hooke007** [reply]
如果你是指 mpv 相关的项目的话，无。

我倾向把所有记录留在 github 可以记录和翻阅
